api/src/main/java/org/openmrs/Obs.java

```diff
@@ -106,49 +132,64 @@
 	 * 
 	 * @see #isObsGrouping() (??)
 	 */
+	@JoinColumn(name = "obs_group_id")
+	@ManyToOne
 	protected Obs obsGroup;
 	
 	/**
 	 * The list of obs grouped under this obs.
 	 */
 	@AllowDirectAccess
+	@JoinColumn(name = "obs_group_id")
+	@OneToMany(cascade = CascadeType.REMOVE)
+	@OrderBy("obs_id")
+	@BatchSize(size = 25)
 	protected Set<Obs> groupMembers;
-	
+	@JoinColumn(name = "value_coded")
+	@ManyToOne
 	protected Concept valueCoded;
-	
+	@JoinColumn(name = "value_coded_name_id")
+	@ManyToOne
 	protected ConceptName valueCodedName;
-	
+	@JoinColumn(name = "value_drug")
+	@ManyToOne
 	protected Drug valueDrug;
-	
+	@Column(name = "value_group_id", length = 11)
```

Comment:
Length is redundant for some types, I would only add it for Date fields and Strings where you want it to be different from the default value of 255

Reply:
i was trying to maintain the defaults used in the hbm.xml file